src/util/http/websocket/MessageSender.h

```diff
@@ -19,30 +19,34 @@ using unique_cleanup::UniqueCleanup;
 /// `QueryToSocketDistributor` of the passed `QueryHub` reference
 /// and provides a generic operator() interface to call addQueryStatusUpdate()
 /// from the non-asio world.
-class UpdateWrapper {
-  UniqueCleanup<std::shared_ptr<QueryToSocketDistributor>> distributor_;
+class MessageSender {
+ private:
+  /// Keep the OwningQueryId alive until distributor_->signalEnd() is called
+  struct DistributorWithFixedLifetime {
+    std::shared_ptr<QueryToSocketDistributor> distributor_;
+    OwningQueryId owningQueryId_;
+
+    DistributorWithFixedLifetime(
+        std::shared_ptr<QueryToSocketDistributor> distributor,
+        OwningQueryId owningQueryId)
+        : distributor_{std::move(distributor)},
+          owningQueryId_{std::move(owningQueryId)} {}
+  };
```

Comment:
```suggestion
  struct DistributorWithFixedLifetime {
    std::shared_ptr<QueryToSocketDistributor> distributor_;
    OwningQueryId owningQueryId_;
  };
```
I see no reason why an (implicitly move-only) aggregate shouldn't work here.

Reply:
And rename it to `DistributorAndOwningQueryId`.

Reply:
> I see no reason why an (implicitly move-only) aggregate shouldn't work here.

It does in theory, but there seems to be a bug in gcc 11 and 12 (didn't bother checking 10), which I mentioned to you a couple of days ago, where this leads to a segmentation fault. I think it might be related to the issue you reported a while ago: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=103909
It also has problems with coroutines and aggregate initialized objects.